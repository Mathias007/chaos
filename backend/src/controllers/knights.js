import { BuildingsSchema, UnitsSchema } from "../models/knights";

import resStatuses from "../config/resStatuses";

const { UNAUTHORIZED, NOT_FOUND } = resStatuses;

exports.getBuildingsList = (req, res, next) => {
    const messages = {
        CASE_UNAUTHORIZED_MESSAGE:
            "Wystąpił problem z autoryzacją przy pobieraniu listy budynków Knights and Merchants.",
        CASE_NOT_FOUND_MESSAGE:
            "Nie znaleziono listy budynków Knights and Merchants.",
        CASE_SUCCESS_MESSAGE:
            "Lista budynków Knights and Merchants została znaleziona."
    };

    const {
        CASE_UNAUTHORIZED_MESSAGE,
        CASE_NOT_FOUND_MESSAGE,
        CASE_SUCCESS_MESSAGE
    } = messages;

    BuildingsSchema.find({}, {}, (err, buildings) => {
        if (err || !buildings) {
            res.status(UNAUTHORIZED).send({
                message: CASE_UNAUTHORIZED_MESSAGE
            });
            next(err);
        } else if (!buildings) {
            res.status(NOT_FOUND).send({
                message: CASE_NOT_FOUND_MESSAGE
            });
        } else {
            res.json({ message: CASE_SUCCESS_MESSAGE, buildings });
        }
    });
};

exports.getUnitsList = (req, res, next) => {
    const messages = {
        CASE_UNAUTHORIZED_MESSAGE:
            "Wystąpił problem z autoryzacją przy pobieraniu listy jednostek Knights and Merchants.",
        CASE_NOT_FOUND_MESSAGE:
            "Nie znaleziono listy jednostek Knights and Merchants.",
        CASE_SUCCESS_MESSAGE:
            "Lista jednostek Knights and Merchants została znaleziona."
    };

    const {
        CASE_UNAUTHORIZED_MESSAGE,
        CASE_NOT_FOUND_MESSAGE,
        CASE_SUCCESS_MESSAGE
    } = messages;

    UnitsSchema.find({}, {}, (err, units) => {
        if (err || !units) {
            res.status(UNAUTHORIZED).send({
                message: CASE_UNAUTHORIZED_MESSAGE
            });
            next(err);
        } else if (!units) {
            res.status(NOT_FOUND).send({
                message: CASE_NOT_FOUND_MESSAGE
            });
        } else {
            res.json({ message: CASE_SUCCESS_MESSAGE, units });
        }
    });
};
