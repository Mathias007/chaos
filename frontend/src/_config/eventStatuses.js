const eventStatuses = {
    articles: {
        ARTICLE_SUCCESSFULLY_LOADED: "ARTICLE_SUCCESSFULLY_LOADED",
        ARTICLE_NOT_FOUND: "ARTICLE_NOT_FOUND",
        ARTS_LIST_LOADED: "ARTS_LIST_LOADED",
        ARTS_LIST_NOT_FOUND: "ARTS_LIST_NOT_FOUND",
        ARTICLE_ADDED: "ARTICLE_ADDED",
        ARTICLE_ADDING_FAILED: "ARTICLE_ADDING_FAILED",
        ARTICLE_SUCCESFULLY_EDITED: "ARTICLE_SUCCESFULLY_EDITED",
        ARTICLE_EDITING_FAILED: "ARTICLE_EDITING_FAILED",
        ARTICLE_SUCCESFULLY_DELETED: "ARTICLE_SUCCESFULLY_DELETED",
        ARTICLE_DELETING_FAILED: "ARTICLE_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    },
    auth: {
        PASSWORD_TOO_SHORT: "PASSWORD_TOO_SHORT",
        PASSWORD_INCORRECT: "PASSWORD_INCORRECT",
        PASSWORDS_NOT_MATCH: "PASSWORDS_NOT_MATCH",
        USER_LOADED: "USER_LOADED",
        LOGIN_SUCCESSFUL: "LOGIN_SUCCESSFUL",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        LOGIN_FAILED: "LOGIN_FAILED",
        LOGOUT_SUCCESSFUL: "LOGOUT_SUCCESSFUL",
        RESET_STATUS: "RESET_STATUS"
    },
    news: {
        NEWS_SUCCESSFULLY_LOADED: "NEWS_SUCCESSFULLY_LOADED",
        NEWS_NOT_FOUND: "NEWS_NOT_FOUND",
        NEWS_LIST_LOADED: "NEWS_LIST_LOADED",
        NEWS_LIST_NOT_FOUND: "NEWS_LIST_NOT_FOUND",
        NEWS_ADDED: "NEWS_ADDED",
        NEWS_ADDING_FAILED: "NEWS_ADDING_FAILED",
        NEWS_SUCCESFULLY_EDITED: "NEWS_SUCCESFULLY_EDITED",
        NEWS_EDITING_FAILED: "NEWS_EDITING_FAILED",
        NEWS_SUCCESFULLY_DELETED: "NEWS_SUCCESFULLY_DELETED",
        NEWS_DELETING_FAILED: "NEWS_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    },
    knights: {
        KAM_BUILDINGS_LOADED: "KAM_BUILDINGS_LOADED",
        KAM_BUILDINGS_NOT_FOUND: "KAM_BUILDINGS_NOT_FOUND",
        KAM_UNITS_LOADED: "KAM_UNITS_LOADED",
        KAM_UNITS_NOT_FOUND: "KAM_UNITS_NOT_FOUND",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR"
    },
    tzar: {
        TZAR_NATIONS_LOADED: "TZAR_NATIONS_LOADED",
        TZAR_NATIONS_NOT_FOUND: "TZAR_NATIONS_NOT_FOUND",
        TZAR_UNITS_LOADED: "TZAR_UNITS_LOADED",
        TZAR_UNITS_NOT_FOUND: "TZAR_UNITS_NOT_FOUND",
        TZAR_SPELLS_LOADED: "TZAR_SPELLS_LOADED",
        TZAR_SPELLS_NOT_FOUND: "TZAR_SPELLS_NOT_FOUND",
        TZAR_TECHNOLOGIES_LOADED: "TZAR_TECHNOLOGIES_LOADED",
        TZAR_TECHNOLOGIES_NOT_FOUND: "TZAR_TECHNOLOGIES_NOT_FOUND",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR"
    },
    users: {
        USER_SUCCESSFULLY_LOADED: "USER_SUCCESSFULLY_LOADED",
        USER_NOT_FOUND: "USER_NOT_FOUND",
        USERS_LIST_LOADED: "USERS_LIST_LOADED",
        USERS_LIST_NOT_FOUND: "USERS_LIST_NOT_FOUND",
        USER_ADDED: "USER_ADDED",
        USER_ADDING_FAILED: "USER_ADDING_FAILED",
        USER_SUCCESFULLY_EDITED: "USER_SUCCESFULLY_EDITED",
        USER_EDITING_FAILED: "USER_EDITING_FAILED",
        USER_SUCCESFULLY_DELETED: "USER_SUCCESFULLY_DELETED",
        USER_DELETING_FAILED: "USER_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    }
};

export default eventStatuses;
