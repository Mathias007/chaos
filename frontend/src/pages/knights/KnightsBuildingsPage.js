import React, { Component } from "react";

import allocationKeys from "../../_config/allocationKeys";
import navigationTitles from "../../_config/navigationTitles";

import BreadcrumbComponent from "../global/BreadcrumbComponent";
import PageHeaderComponent from "../components/PageHeaderComponent";
import PageContentComponent from "../components/PageContentComponent";
import KnightsBuildingsTable from "./data/KnightsBuildingsTable";

import { Layout } from "antd";
import styles from "../../styles/styles";

const { KNIGHTS_AND_MERCHANTS } = navigationTitles;
const { KNIGHTS_BUILDINGS } = allocationKeys;

class KnightsBuildingsPage extends Component {
    state = {
        allocationKey: KNIGHTS_BUILDINGS
    };
    render() {
        return (
            <Layout style={styles.layout}>
                <BreadcrumbComponent
                    section={KNIGHTS_AND_MERCHANTS}
                    page={navigationTitles.KNIGHTS_BUILDINGS}
                />
                <PageHeaderComponent allocationKey={this.state.allocationKey} />
                <PageContentComponent
                    allocationKey={this.state.allocationKey}
                />
                <KnightsBuildingsTable />
            </Layout>
        );
    }
}

export default KnightsBuildingsPage;
