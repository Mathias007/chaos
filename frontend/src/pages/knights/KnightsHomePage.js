import React, { Component } from "react";

import allocationKeys from "../../_config/allocationKeys";

import BreadcrumbComponent from "../global/BreadcrumbComponent";
import PageHeaderComponent from "../components/PageHeaderComponent";
import PageContentComponent from "../components/PageContentComponent";

import { Layout } from "antd";

const { KNIGHTS_HOME } = allocationKeys;

class KnightsHomePage extends Component {
    state = {
        allocationKey: KNIGHTS_HOME
    };

    render() {
        return (
            <Layout style={{ padding: "0 24px 24px" }}>
                <BreadcrumbComponent />
                <PageHeaderComponent allocationKey={this.state.allocationKey} />
                <PageContentComponent allocationKey={this.state.allocationKey} />
            </Layout>
        );
    }
}

export default KnightsHomePage;
