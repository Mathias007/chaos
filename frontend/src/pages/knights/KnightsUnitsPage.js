import React, { Component } from "react";

import KnightsUnitsContent from "./components/KnightsUnitsContent";

class KnightsUnitsPage extends Component {
    state = {};
    render() {
        return <KnightsUnitsContent />;
    }
}

export default KnightsUnitsPage;
