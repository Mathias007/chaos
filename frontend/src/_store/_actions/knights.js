import eventStatuses from "../../_config/eventStatuses";

import fetchBuildingsList from "./services/KnightsBuildingsService";
import fetchUnitsList from "./services/KnightsUnitsService";

const { KAM_BUILDINGS_LOADED, KAM_UNITS_LOADED } = eventStatuses.knights;

export const showBuildings = () => {
    return (dispatch, getState) => {
        const dispatchBuildingsListLoaded = function (buildings) {
            dispatch({
                type: KAM_BUILDINGS_LOADED,
                data: buildings
            });
        };

        return fetchBuildingsList(dispatchBuildingsListLoaded);
    };
};

export const showUnits = () => {
    return (dispatch, getState) => {
        const dispatchUnitsListLoaded = function (units) {
            dispatch({
                type: KAM_UNITS_LOADED,
                data: units
            });
        };

        return fetchUnitsList(dispatchUnitsListLoaded);
    };
};
