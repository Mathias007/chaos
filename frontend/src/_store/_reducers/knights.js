import eventStatuses from "../../_config/eventStatuses";

const {
    KAM_BUILDINGS_LOADED,
    KAM_BUILDINGS_NOT_FOUND,
    KAM_UNITS_LOADED,
    KAM_UNITS_NOT_FOUND,
    AUTHENTICATION_ERROR
} = eventStatuses.knights;

const initialState = {
    buildings: [],
    units: [],
    errorMessage: ""
};

export default function knights(state = initialState, action) {
    switch (action.type) {
        case KAM_BUILDINGS_LOADED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                buildings: action.data.buildings
            };

        case KAM_BUILDINGS_NOT_FOUND:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case KAM_UNITS_LOADED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                units: action.data.units
            };

        case KAM_UNITS_NOT_FOUND:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case AUTHENTICATION_ERROR:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        default:
            return state;
    }
}
