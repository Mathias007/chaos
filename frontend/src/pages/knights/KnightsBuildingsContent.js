import React, { Component } from "react";
import { connect } from "react-redux";
import { knights } from "../../_store/_actions";

import { Layout, Table } from "antd";
import BreadcrumbComponent from "../components/BreadcrumbGlobalComponent";
import Img from "react-image";

// eslint-disable-next-line
const textBetweenTagsRegEx = /[^<\}]+(?=>)/g;

const componentStyles = {
    content: {
        background: "#fff",
        padding: 24,
        margin: 0,
        minHeight: 280
    },
    layout: { padding: "0 24px 24px" }
};

const componentClassnames = {
    images: {
        building: "knights-image-building",
        icon: "knights-image-icon",
        material: "knights-image-material",
        unit: "knights-image-unit"
    },
    content: "knights-buildings-content",
    layout: "knights-buildings-layout",
    table: "knights-buildings-table"
};

class KnightsBuildingsContent extends Component {
    state = {
        imgPath: {
            general: "img",
            section: {
                knights: "knights"
            },
            dir: {
                buildings: "budynki",
                icons: "ikony",
                materials: "surowce",
                units: "jednostki"
            },
            format: {
                bmp: "bmp",
                png: "png",
                PNG: "PNG"
            }
        },
        columnsStructure: {
            col_name: { title: "Nazwa", dataIndex: "name", align: "center" },
            col_cost: { title: "Koszt", dataIndex: "cost", align: "left" },
            col_fields: { title: "Pola", dataIndex: "fields", align: "center" },
            col_durability: {
                title: "Wytrzymałość",
                dataIndex: "durability",
                align: "center"
            },
            col_description: { title: "Działanie", dataIndex: "description" },
            col_worker: {
                title: "Pracownik",
                dataIndex: "worker",
                align: "center"
            },
            col_image: { title: "Grafika", dataIndex: "image", align: "center" }
        },
        descriptionSubHeaders: ["Opis: ", "Potrzebuje: ", "Produkuje: "],
        tableColumns: [],
        tableData: []
    };

    componentDidMount() {
        this.props.showBuildings();
    }

    renderBuildings() {
        let { buildings } = this.props;

        if (buildings) {
            return buildings.map((building, index) => {
                const {
                    _id,
                    nazwa,
                    koszt,
                    pola,
                    wytrzymalosc,
                    dzialanie,
                    pracownik,
                    grafika
                } = building;
                const { deski, kamienie } = koszt;
                const { opis, potrzeby, produkcja } = dzialanie;
                return {
                    key: _id,
                    name: [nazwa, grafika],
                    cost: [deski, kamienie],
                    fields: pola,
                    durability: wytrzymalosc,
                    description: [opis, potrzeby, produkcja],
                    worker: pracownik,
                    image: grafika
                };
            });
        }
    }

    render() {
        const { Content } = Layout;

        let { tableColumns, tableData } = this.state;

        const { general, section, dir, format } = this.state.imgPath;

        tableData = this.renderBuildings();

        const {
            col_name,
            col_cost,
            col_fields,
            col_durability,
            col_description,
            col_worker,
            col_image
        } = this.state.columnsStructure;

        tableColumns = [
            {
                title: col_name.title,
                dataIndex: col_name.dataIndex,
                align: col_name.align,
                render: name => (
                    <>
                        <h3>
                            <strong>{name[0]}</strong>
                        </h3>
                        <Img
                            className={componentClassnames.images.icon}
                            src={require(`../../${general}/${section.knights}/${dir.icons}/${name[1]}.${format.png}`)}
                        />
                    </>
                )
            },
            {
                title: col_cost.title,
                dataIndex: col_cost.dataIndex,
                align: col_cost.align,
                render: cost => {
                    return (
                        <p>
                            {cost[0]}{" "}
                            <Img
                                className={componentClassnames.images.material}
                                src={require(`../../${general}/${section.knights}/${dir.materials}/deska.${format.png}`)}
                            />
                            <br />
                            {cost[1]}
                            <Img
                                className={componentClassnames.images.material}
                                src={require(`../../${general}/${section.knights}/${dir.materials}/kamien.${format.png}`)}
                            />
                        </p>
                    );
                }
            },
            {
                title: col_fields.title,
                dataIndex: col_fields.dataIndex,
                align: col_fields.align
            },
            {
                title: col_durability.title,
                dataIndex: col_durability.dataIndex,
                align: col_durability.align
            },
            {
                title: col_description.title,
                dataIndex: col_description.dataIndex,
                render: description => {
                    const { descriptionSubHeaders } = this.state;
                    return (
                        <span>
                            {description.map((fragment, index) => {
                                if (fragment) {
                                    let materialElements = fragment.match(
                                        textBetweenTagsRegEx
                                    );
                                    return (
                                        <p key={fragment}>
                                            <strong>
                                                {descriptionSubHeaders[index]}
                                            </strong>
                                            {materialElements
                                                ? materialElements.map(
                                                      (element, index) => (
                                                          <Img key={element}
                                                              className={
                                                                  componentClassnames
                                                                      .images
                                                                      .material
                                                              }
                                                              src={require(`../../${general}/${section.knights}/${dir.materials}/${element}.${format.png}`)}
                                                          />
                                                      )
                                                  )
                                                : fragment}
                                        </p>
                                    );
                                } else return null;
                            })}
                        </span>
                    );
                }
            },
            {
                title: col_worker.title,
                dataIndex: col_worker.dataIndex,
                align: col_worker.align,
                render: worker => {
                    let workersElements = worker.match(textBetweenTagsRegEx);
                    console.log(workersElements);
                    return workersElements ? (
                        <Img
                            className={componentClassnames.images.unit}
                            src={require(`../../${general}/${section.knights}/${dir.units}/${workersElements}.${format.PNG}`)}
                        />
                    ) : null;
                }
            },
            {
                title: col_image.title,
                dataIndex: col_image.dataIndex,
                align: col_image.align,
                render: image => (
                    <Img
                        className={componentClassnames.images.building}
                        width="120px"
                        height="100px"
                        src={require(`../../${general}/${section.knights}/${dir.buildings}/${image}.${format.bmp}`)}
                    />
                )
            }
        ];

        return (
            <Layout
                className={componentClassnames.layout}
                style={componentStyles.layout}
            >
                <BreadcrumbComponent />
                <Content
                    className={componentClassnames.images.content}
                    style={componentStyles.content}
                >
                    <h1>Knights and Merchants</h1>
                    <h2>Budynki</h2>

                    <Table
                        className={componentClassnames.table}
                        dataSource={tableData}
                        columns={tableColumns}
                    />
                </Content>
            </Layout>
        );
    }
}

const mapStateToProps = state => {
    return {
        buildings: state.knights.buildings
    };
};

const mapDispatchToProps = dispatch => {
    return {
        showBuildings: () => dispatch(knights.showBuildings())
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(KnightsBuildingsContent);
