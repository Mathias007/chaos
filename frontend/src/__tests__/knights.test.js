import eventStatuses from "../_config/eventStatuses";
import knights from "../_store/_reducers/knights";

const {
    KAM_BUILDINGS_LOADED,
    KAM_BUILDINGS_NOT_FOUND,
    KAM_UNITS_LOADED,
    KAM_UNITS_NOT_FOUND,
    AUTHENTICATION_ERROR
} = eventStatuses.knights;

const initialState = {
    buildings: [],
    units: [],
    errorMessage: ""
};

describe("test users reducer", () => {
    const knightsList = {
        message: "Pozytywna wiadomość z serwera!",
        data: [
            {
                _id: "1111-2222-3333-4444",
                name: "Hobok",
                cost: "1488 gold",
                hp: "500",
                worker: "tester"
            }
        ]
    };

    const message = "Ważna wiadomość z serwera!";

    it("should return the initial state", () => {
        expect(knights(undefined, {})).toEqual(initialState);
    });

    it("should load buildings list", () => {
        const action = {
            type: KAM_BUILDINGS_LOADED,
            data: knightsList.data
        };
        const expectedState = {
            ...initialState,
            ...action.data,
            buildings: action.data.knightsList
        };
        expect(knights(initialState, action)).toEqual(expectedState);
    });

    it("should get message about buildings list's not finding", () => {
        const action = { type: KAM_BUILDINGS_NOT_FOUND, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(knights(initialState, action)).toEqual(expectedState);
    });

    it("should load units list", () => {
        const action = { type: KAM_UNITS_LOADED, data: knightsList.data };
        const expectedState = {
            ...initialState,
            ...action.data,
            units: action.data.knightsList
        };
        expect(knights(initialState, action)).toEqual(expectedState);
    });

    it("should get message about units list's not finding", () => {
        const action = { type: KAM_UNITS_NOT_FOUND, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(knights(initialState, action)).toEqual(expectedState);
    });

    it("should get message about a problem with authentication", () => {
        const action = { type: AUTHENTICATION_ERROR, data: message };
        const expectedState = {
            ...initialState,
            ...action.data,
            errorMessage: action.data.message
        };
        expect(knights(initialState, action)).toEqual(expectedState);
    });
});
