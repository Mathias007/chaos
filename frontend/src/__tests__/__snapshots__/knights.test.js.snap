// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test users reducer should get message about a problem with authentication 1`] = `
Object {
  "buildings": Array [],
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "units": Array [],
}
`;

exports[`test users reducer should get message about buildings list's not finding 1`] = `
Object {
  "buildings": Array [],
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "units": Array [],
}
`;

exports[`test users reducer should get message about units list's not finding 1`] = `
Object {
  "buildings": Array [],
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "units": Array [],
}
`;

exports[`test users reducer should load buildings list 1`] = `
Object {
  "0": Object {
    "_id": "1111-2222-3333-4444",
    "cost": "1488 gold",
    "hp": "500",
    "name": "Hobok",
    "worker": "tester",
  },
  "buildings": undefined,
  "errorMessage": "",
  "units": Array [],
}
`;

exports[`test users reducer should load units list 1`] = `
Object {
  "0": Object {
    "_id": "1111-2222-3333-4444",
    "cost": "1488 gold",
    "hp": "500",
    "name": "Hobok",
    "worker": "tester",
  },
  "buildings": Array [],
  "errorMessage": "",
  "units": undefined,
}
`;
